spras/config/config.py

```diff
@@ -181,35 +146,36 @@ def process_config(self, raw_config):
         # Maps from the dataset label to the dataset list index
         # dataset_dict = {dataset.get('label', f'dataset{index}'): index for index, dataset in enumerate(datasets)}
 
-        # Override the default parameter hash length if specified in the config file
-        if "hash_length" in raw_config and raw_config["hash_length"] != "":
-            self.hash_length = int(raw_config["hash_length"])
-
+    def process_algorithms(self, raw_config: RawConfig):
+        """
+        Parse algorithm information
+        Each algorithm's parameters are provided as a list of dictionaries
+        Defaults are handled in the Python function or class that wraps
+        running that algorithm
+        Keys in the parameter dictionary are strings
+        """
         prior_params_hashes = set()
-
-        # Parse algorithm information
-        # Each algorithm's parameters are provided as a list of dictionaries
-        # Defaults are handled in the Python function or class that wraps
-        # running that algorithm
-        # Keys in the parameter dictionary are strings
         self.algorithm_params = dict()
         self.algorithm_directed = dict()
-        self.algorithms = raw_config["algorithms"]
+        self.algorithms = raw_config.algorithms
         for alg in self.algorithms:
-            cur_params = alg["params"]
-            if "include" in cur_params and cur_params.pop("include"):
+            cur_params = alg.params
+            if cur_params.include:
                 # This dict maps from parameter combinations hashes to parameter combination dictionaries
-                self.algorithm_params[alg["name"]] = dict()
+                self.algorithm_params[alg.name] = dict()
             else:
                 # Do not parse the rest of the parameters for this algorithm if it is not included
                 continue
 
-            if "directed" in cur_params:
-                print("UPDATE: we no longer use the directed key in the config file")
-                cur_params.pop("directed")
+            if cur_params.directed is not None:
+                warnings.warn("UPDATE: we no longer use the directed key in the config file", stacklevel=2)
+
+            cur_params = cur_params.__pydantic_extra__
+            if cur_params is None:
+                raise RuntimeError("An internal error occured: ConfigDict extra should be set on AlgorithmParams.")
```

Comment:
```suggestion
                raise RuntimeError("An internal error occurred: ConfigDict extra should be set on AlgorithmParams.")
```

This message reads as if it is aimed at developers. Will a user know what to do?

Reply:
A user will not know what to do. However, I can not see a scenario where this would not be caught in testing, since ConfigDict is set in AlgorithmParams.